scripts/staging/release-existence.sh

```diff
@@ -0,0 +1,180 @@
+
+echo "Tag: $TAG"
+echo "Version: $VERSION"
+echo "Type: $TYPE"
+echo "Github Organization: ${GITHUB_ORG}"
+echo "Dockerhub Organization: ${DOCKERHUB_ORG}"
+echo "Images to check: ${IMAGES[*]}"
+echo "Registries: ${IMAGE_REGISTRY}"
+echo "Chart: ${CHART_NAME}"
+echo "Chart Location: ${CHART_REGISTRY}"
+echo ""
+
+VALIDATION_FAILED=false
+
+check_github_tag() {
+    if git ls-remote -t "${INDEX_REMOTE}" | grep -q "refs/tags/${TAG}$"; then
+        log_fatal "Tag ${TAG} exists on remote ${INDEX_REMOTE}"
+    else
+        echo "Tag ${TAG} does not exist on remote ${INDEX_REMOTE}"
+    fi
+}
+
+dockerhub_tag_exists() {
+    local repository="$1"
+    local tag="$2"
+
+    if ! curl --silent -f -lSL "https://hub.docker.com/v2/repositories/${repository#docker.io/}/tags/${tag}" 1>/dev/null 2>&1; then
+        if [[ "$repository" =~ ':' ]]; then
+            curl --silent -f -lSL "http://${repository%%/*}/v2/${repository#*/}/manifests/${tag}" 1>/dev/null 2>&1
+        else
+            return 1
+        fi
+    fi
+}
+
+check_docker_image() {
+    local registry="$1"
+    local org="$2"
+    local image="$3"
+    local tag="$4"
+
+    if [[ "$registry" == "docker.io" ]] ; then
+        if dockerhub_tag_exists "${org}/${image}" "${tag}"; then
+            log_fatal "Image ${org}/${image}:${tag} already exists on Docker Hub"
+        else
+            echo "Image ${org}/${image}:${tag} does not exist on Docker Hub"
+        fi
+    else
+        log_fatal "Unknown registry: $registry"
+    fi
+}
+
+check_helm_chart_oci() {
+    local registry="$1"
+    local namespace="$2"
+    local chart="$3"
+    local version="$4"
+
+    local ref="${registry}/${namespace}/${chart}"
+
+    if helm show chart "${ref}" --version "${version}" >/dev/null 2>&1; then
+        log_fatal "Helm chart ${ref}:${version} already exists in ${registry}"
+    else
+        echo "Helm chart ${ref}:${version} does not exist in ${registry}"
+    fi
+}
+
+
+index_yaml()
+{
+    git fetch "$INDEX_REMOTE" "$INDEX_BRANCH" --depth 1 2>/dev/null
+    INDEX_FILE_YAML="$(git show "$INDEX_REMOTE"/"$INDEX_BRANCH":"$INDEX_BRANCH_FILE")"
+    echo "$INDEX_FILE_YAML"
+}
+
+check_helm_chart_gh_pages() {
+    INDEX_FILE_YAML=$(index_yaml)
+    len_versions="$(echo "$INDEX_FILE_YAML" | yq ".entries.${CHART_NAME} | length")"
+    INDEX_VERSIONS="$(echo "$INDEX_FILE_YAML" | yq ".entries.${CHART_NAME}[].version")"
+    INDEX_CHART_VERSIONS=$(echo "$INDEX_VERSIONS" | grep "$VERSION" || echo)
+
+    if [ "$(echo "$INDEX_VERSIONS" | grep -x "$VERSION" || echo)" == "$VERSION" ]; then
+        log_fatal "A stable chart version $VERSION matching the app tag is already in the index"
+    else
+        echo "Helm chart ${CHART_NAME}:${VERSION} does not exist on GitHub Pages"
+    fi
+}
+
+if [[ "$TYPE" == "staging" ]]; then
+    check_github_tag
+fi
+
+for image in "${IMAGES[@]}"; do
+    if ! check_docker_image "${IMAGE_REGISTRY}" "${DOCKERHUB_ORG}" "${image}" "v${TAG}"; then
+        VALIDATION_FAILED=true
```

Comment:
we're not doing anything with `VALIDATION_FAILED` ?

Reply:
yes we are right? the final one

Reply:
only after you force-pushed :)

Reply:
Ohh